pkg/apiserver/resthandler.go

```diff
@@ -82,7 +82,13 @@ func GetResource(r RESTGetter, ctxFn ContextFunc, namer ScopeNamer, codec runtim
 }
 
 func parseSelectorQueryParams(query url.Values, version, apiResource string) (label, field labels.Selector, err error) {
-	label, err = labels.Parse(query.Get("labels"))
+	var labelString string
+	if version == "v1beta1" || version == "v1beta2" {
+		labelString = query.Get("labels")
+	} else {
+		labelString = query.Get("label-selector")
```

Comment:
ok, moved this to a function in ```api``` and refactored all accesses in terms of a versioned request to that function.